codegen/lib/objc-codegen.js

```
@@ -0,0 +1,280 @@
+
+  var reverseMethod = Object.create(method);
+  reverseMethod.name = reverseName;
+  reverseMethod.internal = 'Use ' + ngModelName + '.' + apiName + '() instead.';
+  // override possibly inherited values
+  reverseMethod.deprecated = false;
+
+  var reverseModel = findModelByName(models, targetClass);
+  reverseModel.methods.push(reverseMethod);
+  if(reverseMethod.name.match(/create/)){
+    var createMany = Object.create(reverseMethod);
+    createMany.name = createMany.name.replace(
+      /create/,
+      'createMany'
+    );
+    createMany.internal = createMany.internal.replace(
+      /create/,
+      'createMany'
+    );
+    createMany.isReturningArray = function() { return true; };
+    reverseModel.methods.push(createMany);
+  }
+
+  var scopeMethod = Object.create(method);
+  scopeMethod.name = reverseName;
+  // override possibly inherited values
+  scopeMethod.deprecated = false;
+  scopeMethod.internal = false;
+  modelClass.scopes[scopeName].methods[apiName] = scopeMethod;
+  if(scopeMethod.name.match(/create/)){
+    var scopeCreateMany = Object.create(scopeMethod);
+    scopeCreateMany.name = scopeCreateMany.name.replace(
+      /create/,
+      'createMany'
+    );
+    scopeCreateMany.isReturningArray = function() { return true; };
+    apiName = apiName.replace(/create/, 'createMany');
+    modelClass.scopes[scopeName].methods[apiName] = scopeCreateMany;
+  }
+}
+
+function findModelByName(models, name) {
+  for (var n in models) {
+    if (n.toLowerCase() == name.toLowerCase())
+      return models[n];
+  }
+}
+
+function addObjCNames(models, modelPrefix) {
+  for (var modelName in models) {
+    var meta = models[modelName];
+    meta.objcModelName = modelPrefix + modelName[0].toUpperCase() + modelName.slice(1);
+    if (meta.baseModel === 'Model' || meta.baseModel === 'PersistedModel') {
+      meta.objcBaseModel = 'LB' + meta.baseModel;
+    } else {
+      console.error('unknown baseModel name: ' + meta.baseModel); // FIXME
```

I think you can simply throw an error:

```js
throw new Error('...');
```

Please include `modelName` in the error message too.